src/models/PlantModel.ts

```diff
@@ -1,74 +1,83 @@
-import { HandleFile, FileType } from './HandleFile';
-import { IPlant, IOpsInfo } from '../interfaces';
+import { ResultSetHeader } from 'mysql2';
+import connection from './connection';
+import { IPlant } from '../interfaces';
 
 class PlantModel {
-  private handleFile = new HandleFile();
-
-  private async updateOpsInfo(incrementAmount = 1): Promise<number> {
-    const opsInfo = await this.handleFile.readFile<IOpsInfo>(FileType.OpsInfo);
-    opsInfo.createdPlants += incrementAmount;
-
-    await this.handleFile.saveFile(FileType.OpsInfo, opsInfo);
-
-    return opsInfo.createdPlants;
-  }
+  private conn = connection;
 
   public async getAll(): Promise<IPlant[]> {
-    const plants = await this.handleFile.readFile<IPlant[]>(FileType.Plants);
+    const query = `SELECT
+      id, breed, size, needs_sun as needsSun, origin, water_Frequency as waterFrequency
+      FROM plants`;
+    const [rows] = await this.conn.execute(query);
+    const plants = rows as IPlant[];
     return plants;
   }
 
   public async create(plant: Omit<IPlant, 'id'>): Promise<IPlant> {
-    const plants = await this.getAll();
-
-    const newPlantId = await this.updateOpsInfo(1);
-    const newPlant = { id: newPlantId, ...plant };
-    plants.push(newPlant);
-
-    await this.handleFile.saveFile(FileType.Plants, plants);
-
+    const {
+      breed, needsSun, origin, size, waterFrequency,
+    } = plant;
+    const query = `INSERT INTO plants (breed, needs_sun, origin, size, water_frequency)
+      VALUES (?, ?, ?, ?, ?)`;
+    const values = [breed, needsSun, origin, size, waterFrequency];
+
+    const [rows] = await this.conn.execute<ResultSetHeader>(query, values);
+
+    const newPlant = {
+      id: rows.insertId,
```

Comment:
Aqui chamar de `rows` não faz sentido, dado que o `ResultSetHeader` não retorna `rows`, só o resultado da operação de insert. Não gosto muito de `result` pq bem genérico, mas ainda é mais acurado que `rows`.

```suggestion
    const [result] = await this.conn.execute<ResultSetHeader>(query, values);

    const newPlant = {
      id: result.insertId,
```

Reply:
Alteração implementada